art/attacks/inference/membership_inference/label_only_boundary_distance.py

```diff
@@ -70,6 +71,18 @@ def infer(self, x: np.ndarray, y: Optional[np.ndarray] = None, **kwargs) -> np.n
         """
         from art.attacks.evasion.hop_skip_jump import HopSkipJump
 
+        if y is None:
```

Comment:
Why not make the parameter non-optional?

Reply:
This to follow the overall pattern in ART. For membership inference attacks we could discuss to remove the Optional in general as it is not supported by any of the current membership inference attacks.